OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java

```diff
@@ -285,11 +285,7 @@ public boolean isMainUnitLoaded() {
 
     @Override
     public boolean showGoogleCastButton() {
-        if (pager2 != null && pagerAdapter != null) {
-            CourseUnitFragment fragment = ((CourseUnitFragment) pagerAdapter.getItem(pager2.getCurrentItem()));
-            return fragment != null && fragment.hasCastSupportedVideoContent();
-        }
-        return super.showGoogleCastButton();
+        return (selectedUnit != null && selectedUnit.getType() == BlockType.VIDEO);
```

Comment:
there should be some checks like the following.
```java
    @Override
    public boolean showGoogleCastButton() {
        VideoBlockModel videoBlock = null;
        if (selectedUnit instanceof VideoBlockModel) {
            videoBlock = (VideoBlockModel) selectedUnit;
        }
        boolean isYoutubeVideo = videoBlock != null && videoBlock.getData().encodedVideos.getYoutubeVideoInfo() != null;
        return (selectedUnit != null && selectedUnit.getType() == BlockType.VIDEO && !isYoutubeVideo);
    }
```

cuz app doesn't support cast feature for **Youtube** videos for now.